download.py

```diff
@@ -0,0 +1,95 @@
+import json
+import sys
+import os
+from datetime import datetime, date
+import calendar
+try:
+    import urllib.request as urlrequest
+except ImportError:
+    import urllib as urlrequest
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse
+
+
+def main():
+    with open('config.json') as config_file:
+        config = json.load(config_file)
+
+    installations = config['installations']
+    api_response_cache_dir = config['api_response_cache_dir']
+    num_months_to_process = config['num_months_to_process']
+    monthly_endpoints = config['endpoints']['monthly']
+    single_endpoints = config['endpoints']['single']
+
+    for installation in installations:
+        process_monthly_endpoints(installation, monthly_endpoints, api_response_cache_dir, num_months_to_process)
+        process_single_endpoints(installation, single_endpoints, api_response_cache_dir)
+
+
+def process_monthly_endpoints(installation, monthly_endpoints, api_response_cache_dir, num_months_to_process):
+    for endpoint in monthly_endpoints:
+        process_monthly_endpoint(installation, endpoint, api_response_cache_dir, num_months_to_process)
+
+
+def process_monthly_endpoint(installation, endpoint, api_response_cache_dir, num_months_to_process):
+    for month in get_months(num_months_to_process):
+        url = installation + '/api/info/metrics/' + endpoint + '/' + month
+        response = urlrequest.urlopen(url)
+        json_out = get_remote_json(response)
+        o = urlparse(installation)
+        hostname = o.hostname
+        path = api_response_cache_dir + '/' + endpoint + '/' + month
+        if not os.path.exists(path):
+            os.makedirs(path)
+        filename = hostname + '.json'
+        with open(path + '/' + filename, 'w') as outfile:
+            json.dump(json_out, outfile, indent=4)
+
+
+def get_months(num_months_to_process):
+    months = []
+    today = datetime.today()
+    for i in range(num_months_to_process):
+        months.append(subtract_months(today, i).strftime('%Y-%m'))
+    return months
+
+
+# variation of https://stackoverflow.com/questions/4130922/how-to-increment-datetime-by-custom-months-in-python-without-using-library/4131114#4131114
+def subtract_months(sourcedate, months):
```

Comment:
Minor suggestion: seems you split up other word parameters by snake_casing them, maybe do the same for sourcedate here?

Reply:
Yes. Good point. In the next go around it would be nice to fix this. I don't write a ton of Python but my understanding is that snake_case is preferred.